pkg/components/profiler/profilerservice/service.go

```diff
@@ -26,15 +26,18 @@ import (
 	"github.com/radius-project/radius/pkg/ucp/ucplog"
 )
 
-type Service struct {
-	Options HostOptions
+// Options represents the options for enabling pprof profiler.
+type Options struct {
+	// Enabled is a flag to enable the profiler.
+	Enabled bool `yaml:"enabled,omitempty"`
+
+	// Port is the port on which the profiler server listens.
+	Port int `yaml:"port,omitempty"`
 }
 
-// NewService of profiler package returns a new Service with the configs needed
-func NewService(options HostOptions) *Service {
-	return &Service{
-		Options: options,
-	}
```

Comment:
These services had a few steps of extra boilerplate that weren't accomplishing anything. I simplified them and made them consistent.